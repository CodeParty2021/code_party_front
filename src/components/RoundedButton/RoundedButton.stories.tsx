import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { RoundedButton } from "./RoundedButton";

export default {
  title: "components/RoundedButton",
  component: RoundedButton,
} as ComponentMeta<typeof RoundedButton>;

const Template: ComponentStory<typeof RoundedButton> = (args) => (
  <RoundedButton {...args} />
);

export const Default = Template.bind({});
Default.args = {
  value: "次へ",
  disabled: false,
  icon: null,
};
