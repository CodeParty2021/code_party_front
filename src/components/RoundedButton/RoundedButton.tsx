import { ArrowLeft, ArrowRight } from "components/icons";
import React from "react";
import { roundedButton } from "styles/colors";
import {
  RoundedButtonStyle,
  RoundedButtonStyleProps,
} from "./RoundedButtonStyle";

type Props = RoundedButtonStyleProps & {
  icon?: "left" | "right" | null;
  value?: string;
  disabled?: boolean;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
};

export const RoundedButton: React.FC<Props> = ({
  icon,
  value,
  disabled,
  onClick,
  ...styleProps
}) => {
  return (
    <RoundedButtonStyle
      onClick={onClick}
      disabled={disabled}
      {...styleProps}
      status={disabled ? "disabled" : "default"}
      icon={icon}
    >
      {icon == "left" && <ArrowLeft size={16} fill={roundedButton.font} />}
      <span>{value}</span>
      {icon == "right" && <ArrowRight size={16} fill={roundedButton.font} />}
    </RoundedButtonStyle>
  );
};
