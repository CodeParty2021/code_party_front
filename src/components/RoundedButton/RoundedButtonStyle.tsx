import styled, { css } from "styled-components";
import { roundedButton } from "styles/colors";
import { FlexGap } from "styles/FlexGap/FlexGap";

export type RoundedButtonStyleProps = {};

type LocalStyleProps = {
  status?: "default" | "disabled";
};

const defaultStyle = css`
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  ${FlexGap({ gap: "8px", direction: "row" })}
  padding: 4px 12px 4px 16px;

  background: ${roundedButton.bg};
  border: 1px solid ${roundedButton.border};
  border-radius: 40px;

  color: ${roundedButton.font};
  font-size: 16px;
  line-height: 160%;
`;

const disabledStyle = css`
  & > * {
    opacity: 0.2;
  }
`;

const hoverStyle = css`
  :hover {
    border-color: ${roundedButton.hover.border};
  }
`;

export const RoundedButtonStyle = styled.button<
  RoundedButtonStyleProps & LocalStyleProps
>`
  ${defaultStyle}

  ${({ status }) => status != "disabled" && hoverStyle}
  ${({ status }) => status == "disabled" && disabledStyle}
`;

RoundedButtonStyle.defaultProps = {};
