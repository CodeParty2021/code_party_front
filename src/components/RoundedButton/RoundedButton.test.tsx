import React from "react";
import { RoundedButton } from "./RoundedButton";
import "jest-styled-components";
import { render, shallow } from "enzyme";

describe("<RoundedButton />", () => {
  it("snapshot test", () => {
    const wrapper = render(
      <RoundedButton
        value="次へ"
        icon="left"
        disabled={false}
        onClick={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("render test disabled", () => {
    expect(
      render(
        <RoundedButton
          value="次へ"
          icon="left"
          disabled={true}
          onClick={() => {}}
        />
      )
    ).toBeTruthy();
  });

  it("click test", () => {
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <RoundedButton
        value="次へ"
        icon="left"
        disabled={false}
        onClick={clickHandler}
      />
    );
    wrapper.simulate("click");
    expect(clickHandler).toHaveBeenCalled();
  });
});
