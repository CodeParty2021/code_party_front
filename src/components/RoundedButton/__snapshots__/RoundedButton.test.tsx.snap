// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoundedButton /> snapshot test 1`] = `
.c1 {
  display: inline-block;
  width: 16px;
  height: 16px;
  fill: #363E59;
}

.c1 > span,
.c1 > div,
.c1 > svg {
  display: block;
  width: 100%;
  height: 100%;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
  height: 40px;
  background: #F5F6F8;
  border: 1px solid #D7DAE4;
  border-radius: 40px;
  color: #363E59;
  font-size: 16px;
  line-height: 160%;
  padding: 4px 16px 4px 12px;
}

.c0:hover {
  border-color: #6675FC;
}

@supports not (gap:8px) {
  .c0 > *:not(:first-child) {
    margin-left: 8px;
  }
}

<button
  class="c0"
>
  <span
    class="c1"
    display="inline-block"
    fill="#363E59"
  />
  <span>
    次へ
  </span>
</button>
`;
